tests/v2/validation/Jenkinsfile_support_matrix_os

```diff
@@ -0,0 +1,238 @@
+
+                  echo awsEC2ConfigParam
+
+                  rke1NodeParams = [
+                    string(name: 'TIMEOUT', value: "${env.TIMEOUT}"),
+                    text(name: 'CONFIG', value: awsNodeTemplateParam),
+                    string(name: 'TEST_PACKAGE', value: "${env.RANCHER_RKE1_TEST_PACKAGE}"),
+                    string(name: 'GOTEST_TESTCASE', value: "${env.RANCHER_RKE1_NODE_GOTEST_TESTCASE}"),
+                  ]
+                  rke2NodeParams = [
+                    string(name: 'TIMEOUT', value: "${env.TIMEOUT}"),
+                    text(name: 'CONFIG', value: awsMachineConfigParam),
+                    string(name: 'TEST_PACKAGE', value: "${env.RANCHER_RKE2_TEST_PACKAGE}"),
+                    string(name: 'GOTEST_TESTCASE', value: "${env.RANCHER_RKE2_NODE_GOTEST_TESTCASE}"),
+                  ]
+                  rke1CustomParams = [
+                    string(name: 'TIMEOUT', value: "${env.TIMEOUT}"),
+                    text(name: 'CONFIG', value: awsEC2ConfigParam),
+                    string(name: 'TEST_PACKAGE', value: "${env.RANCHER_RKE1_TEST_PACKAGE}"),
+                    string(name: 'GOTEST_TESTCASE', value: "${env.RANCHER_RKE1_CUSTOM_GOTEST_TESTCASE}"),
+                  ]
+                  rke2CustomParams = [
+                    string(name: 'TIMEOUT', value: "${env.TIMEOUT}"),
+                    text(name: 'CONFIG', value: awsEC2ConfigParam),
+                    string(name: 'TEST_PACKAGE', value: "${env.RANCHER_RKE2_TEST_PACKAGE}"),
+                    string(name: 'GOTEST_TESTCASE', value: "${env.RANCHER_RKE2_CUSTOM_GOTEST_TESTCASE}"),
+                  ]
+
+                  jobs = [:]
+
+                  failBuild = false
+                  failJobs =[]
+
+                  jobs["rke1node"] = { 
+                    try {build job: 'go-validation-provisioning', parameters: rke1NodeParams}
+                    catch (err) {
+                      echo err.toString()
+                      failBuild = true
+                    } 
+                  }
+                  jobs["rke2node"] = { 
+                    try {build job: 'go-validation-provisioning', parameters: rke2NodeParams}
+                    catch (err) {
+                      echo err.toString()
+                      failBuild = true
+                    } 
+                  }
+                  jobs["rke1custom"] = { 
+                    try {build job: 'go-validation-provisioning', parameters: rke1CustomParams}
+                    catch (err) {
+                      echo err.toString()
+                      failBuild = true
+                    } 
+                  }
+                  jobs["rke2custom"] = { 
+                    try {build job: 'go-validation-provisioning', parameters: rke2CustomParams}
+                    catch (err) {
+                      echo err.toString()
+                      failBuild = true
+                    } 
+                  }
+
+                  parallel jobs
```

Comment:
Same question for this one too; is this in the loop?

Reply:
Yes if I remember correctly it runs all provisioning jobs for a given os, and then loops to the next os.

Reply:
Yes, that's correct, it takes a list of OS, creates an array of jobs to run for each os, then runs all the jobs in that array in parallel, then does the same for the next OS in the array.

This way you get OS specific batches of jobs that are easier to parse after the job completes, but I just realized it would be faster to loop through the all the OSs, create all the jobs, then parallel them all at once. I'd definitely want to also add some sort of messaging or reporting on which jobs failed and why beyond what already exists in that case though.

Reply:
I prefer this way, I agree with you it is easier to follow. So let's keep it this way.